plugins/kubernetes-backend/src/service/KubernetesClientProvider.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ClusterDetails } from '../cluster-locator/types';
+import { AppsV1Api, CoreV1Api, KubeConfig } from '@kubernetes/client-node';
+
+export class KubernetesClientProvider {
```

Comment:
I think we'll dive into this interface a bit more when look at auth right? But with per-user auth it wouldn't really be enough with a cluster->client mapping afaik

Reply:
Yes, the auth work will expand all of this further and the interfaces here are not final